test/functions.jl

```diff
@@ -0,0 +1,39 @@
+@testset "closure" begin
+    magic(x) = false
+    sentinel = gensym("sentinel")
+    @test magic(sentinel) == false
+
+    # Getting closers to work means having a function created in the current scope
+    patch = @patch magic(x) = x == sentinel
+    apply(patch) do
+        @test (@mock magic(sentinel)) == true
+    end
+end
+
+# https://github.com/invenia/Mocking.jl/issues/56
+@testset "kwarg default uses arg" begin
+    function f end
+
+    p = @patch function f(x; y=x)
```

Comment:
Does this mean you can patch a function as a different signature?  since the original definition of `f` above didn't have any args?

Reply:
wouldn't that just be defining a new function rather than patching an existing one? or am I not understanding?

Reply:
Patches aren't required to match the signature of an existing method but they do need to target an existing function. I'm creating the empty generic function `f` just to make something to target. I'll note it can be nice for developers not to have to match signatures exactly as this could make patch creation more fastidious than necessary